reproducer_metal_address_space.py

```
@@ -0,0 +1,111 @@
+
+# Step 3: Try to compile → capture source
+print("=== Reproducing the bug ===")
+N, BN, TH = 4096, 256, 128
+A = torch.randn(N, dtype=torch.float32, device="mps")
+B = torch.randn(N, dtype=torch.float32, device="mps")
+
+try:
+    f = vec_add.compile(A, B, block_N=BN, threads=TH)
+    print("  Metal kernel compiled successfully! (bug already fixed)")
+except SyntaxError as e:
+    print(f"  ❌ Bug reproduced!")
```

Ruff (0.15.21)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)